lib/gps.js

```diff
@@ -0,0 +1,394 @@
+
+
+/**
+ *
+ * @constructor
+ *
+ * @param {Object} opts Options: pin(s), chip, receiver, breakout, precision, serialport, freq
+ *
+ * Sample initialization
+ *
+ *    new five.GPS({ pins: {rx: 10, tx: 11});
+ *
+ */
+
+function GPS(opts) {
+
+  var breakout, receiver, chip, state;
+
+  if (!(this instanceof GPS)) {
+    return new GPS(opts);
+  }
+
+  // Allow users to pass in a 2 element array for tx and rx pins
+  if (Array.isArray(opts)) {
+    opts = { pins: {tx: opts[0], rx: opts[1], onOff: opts[2] }};
+  }
+
+  if (typeof opts.pins === "undefined") {
+    opts.pins = {};
+  }
+
+  Board.Component.call(
+    this, this.opts = Board.Options(opts)
+  );
+
+  breakout = opts.breakout || "adafruitUltimateGPS";
+
+  // Get user overrides for chip and receivers
+  receiver = opts.receiver;
+  chip = opts.chip;
+
+  if (!receiver && Breakouts[breakout].receiver) {
+    receiver = Breakouts[breakout].receiver.value;
+  }
+
+  if (!chip && Breakouts[breakout].chip) {
+    chip = Breakouts[breakout].chip.value;
+  }
+
+  // If a receiver was defined or derived but chip was not
+  if (receiver && !chip && Receivers[receiver].chip) {
+    chip = Receivers[receiver].chip.value;
+  }
+
+  // Allow users to pass in custom chip types
+  chip = typeof chip === "string" ?
+    Chips[chip] : this.opts.chip;
+
+  // Allow users to pass in custom receiver types
+  receiver = typeof receiver === "string" ?
+    Receivers[receiver] : this.opts.receiver;
+
+  // Chip decorates the instance
+  Object.defineProperties(this, chip);
+
+  // Receiver decorates this instance
+  Object.defineProperties(this, receiver);
+
+  // breakout decorates the instance
+  if (this.opts.breakout) {
+    breakout = typeof this.opts.breakout === "string" ?
+      Breakouts[this.opts.breakout] : this.opts.breakout;
+
+    Board.Controller.call(this, breakout, opts);
+  }
+
+  // If necessary set default property values
+  this.precision = opts.precision || 9;
+  this.baud = opts.baud || breakout.baud || chip.baud;
+
+  // Create a "state" entry for privately
+  // storing the state of the instance
+  state = {
+    sat: {},
+    pos: {},
+    nav: {},
+    frequency: 1000,
+    lowPowerMode: false
+  };
+
+  priv.set(this, state);
+
+  // Getters for private state values
+  Object.defineProperties(this, {
+    pos: {
+      get: function() {
+        return state.pos;
+      }
+    },
+    sat: {
+      get: function() {
+        return state.sat;
+      }
+    },
+    nav: {
+      get: function() {
+        return state.nav;
+      }
+    },
+    time: {
+      get: function() {
+        return state.time;
+      }
+    }
+  });
+
+  if (this.initialize) {
+    this.initialize(opts);
+  }
+
+}
+
+util.inherits(GPS, events.EventEmitter);
+
+/*
+ * Default intialization for serial GPS
+ */
+GPS.prototype.initialize = function(opts) {
+
+  var state = priv.get(this);
+  state.serialPort = opts.serialPort || opts.portId || opts.port || opts.bus || this.io.SERIAL_PORT_IDs.SW_SERIAL0;
```

Comment:
I was leaning toward SW serial because I've always considered the Uno to be our prototypical use case. So if the number of detected Serial pins === 0 I could assume I'm talking to an Uno or other board with just the one UART and should use SW_SERIAL0. That way Mega users would get HW_SERIAL0 by default and ...

You know what, this is just going to confuse people. I should go back to requiring the port to be specified. It's a little more configuration, but it will be more clear to users what's going on.